src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
maybe it could be even 1 call again: async_response(barrier=False).
- without barrier, it would just fetch one response.
- with barrier, it would fetch one response and even wait for it, if it is not there yet. or return None if there is nothing left.
(just a vague idea, not fully thought through)